docsrc/interactivity.nim

```diff
@@ -2,57 +2,125 @@ import nimib
 
 nbInit
 
+template nimibCode*(body: untyped) =
+  newNbCodeBlock("nimibCode", body):
+    discard
+  body
+
+nb.partials["nimibCode"] = nb.partials["nbCode"]
+nb.renderPlans["nimibCode"] = nb.renderPlans["nbCode"]
+
 nbText: hlMd"""
-# Creating interactive components in Nimib
+# Creating interactive components in nimib
 
 Nimib can easily be used to create static content with `nbText` and `nbCode`, but did you know that you can create interactive
 content as well? And that you can do it all in Nim even! This can be achieved using either the `nbJsFromCode`-API or `nbKaraxCode`.
 They work by compiling Nim code into javascript and adding it to the resulting HTML file.
 This means that arbitrary Javascript can be written but also that Karax, which compiles to javascript, also can be used.
+The available template are:
+- `nbJsFromCode`: compile the code to js.
+- `nbJsFromCodeInBlock`: compile to js and put it inside a `block`.
+- `nbJsFromCodeGlobal`: compile to js and place at the top of the generated file.
+- `nbJsFromCodeOwnFile`: compile to js as its own file. (The blocks above are cumulated into a single big file)
+- `nbKaraxCode`: Sugar on top of `nbJsFromCodeOwnFile` for writing Karax components.
+
+It's important to note that code inside `nbJsFromCode/InBlock/Global` will be pasted into a single file and compiled together.
+This has consequences which will be discussed further below, but the main advantages is that it reduces build times and allows
+for interactions between nbJs-blocks. 
 
-## nbJsFromCodeInit
-This is the fundamental API used for compiling Nim-snippets to javascript. It consists of three templates:
-- `nbJsFromCodeInit` - Creates a new code script that further code can be added to later.
-- `addCodeToJs` - Adds to an existing code script
-- `addToDocAsJs` - Takes the Nim code in a script and compiles it to javascript. 
+## nbJsFromCode
+This is the fundamental API used for compiling Nim-snippets to javascript.
 Here is a basic example:
 """
 
-nbCode:
-  let script = nbJsFromCodeInit:
-    echo "Hello world!"
-  let x = 3.14
-  script.addCodeToJs(x):
-    echo "Pi is roughly ", x
-  ## Uncomment this line:
-  ##script.addToDocAsJs()
-script.addToDocAsJs()
-nbJsShowSource("This is the complete script:")
+nimibCode:
+  nbJsFromCode:
+    let x = "Hello world!"
+    echo x
 
+nbText: hlMd"""
+If you now go to your browser's javascript console you should see `Hello world` printed there.
+So the code we passed to `nbJsFromCode` has been compiled to Javascript and is run by your browser!
 
+### Capturing variables
```

Comment:
at the end of capturing variables section we could point out that capturing variables is essential for a js widget library and refer to counter document to show an example of that.